src/identities/providers/github/GitHubProvider.ts

```diff
@@ -42,7 +42,7 @@ class GitHubProvider extends Provider {
   }
 
   public async *authenticate(
-    timeout: number = 120000,
+    timeout: number = Infinity,
   ): AsyncGenerator<ProviderAuthenticateRequest, IdentityId> {
```

Comment:
I agree, I set it to 1200000 (that is 10x of initial timeout earlier, but you said Infinity would make more sense, I'll change it back to 1200k seconds, which would be, 20 minutes. More than enough.

Reply:
20 minutes is too much, this caller is setting the timeout anyway with the timeout of the `IdentitiesAuthenticate` handler. So the change should be made there rather than in `GithubProvider`.